gitbook-docs/access_requests.md

```diff
@@ -0,0 +1,97 @@
+# Access Requests
+
+When a device needs to gain access to a secured Signal K server, it can use __Access Requests__ to request and be
+granted access to the server.
+
+See [Request/Response](request_response.md) for more information on request/response semantics.
+
+A device could be a display or for example an engine sensor or a temperature sensor.
+
+## Definitions
+
+* `clientId` is a string that uniquely identifies a device. This could be a UUID, a combination of model number and
+  serial number, or any other string. The client is responsible for issuing the value. The client should use the same
+  value for all its requests.
+* `requestId` is a string generated by the server in response to an access request and used by the client to correlate
+  the request with the results. The client should not rely on the format or contents of this string.
+
+## Device Requests
+
+The device will send a REST request to the server:
+
+```sh
+$ curl -k \
+    --header "Content-Type: application/json" \
+    --request POST \
+    --data '{"clientId":"1234-45653-343453","description":"My Awesome Humidity Sensor"}' \
+    https://localhost:3443/signalk/v1/access/requests
+```
+
+If the server does not support access requests it must return HTTP status code 501 Not Implemented.
+
+For a successfully received access request the server must return an HTTP status code 202 and a JSON response with an
+href to check the status and get the response.
+
+```json
+{
+  "href": "/signalk/v1/access/requests/358b5f32-76bf-4b33-8b23-10a330827185"
```

Comment:
I've removed `result` from all the http responses since it is the http response code.

Reply:
wait, I take that back. See below...

Reply:
The result key is for non http requests. We also have to support devices on raw tcp, serial, etc
Strictly speaking it could be stripped from the http response but that will just make complexity for no gain.

Reply:
Let's not get into the specifics of the requests response stuff here. There's is a separate PR for that.

Reply:
Its not clear to me what the logic is for when a `result` should or should not be included in the response. Is it just a mirror of the HTTP result or does it indicate something else?

Reply:
That's clearly stated (I think?) in the request/response document: https://github.com/SignalK/specification/pull/508/files#diff-145e073ee850b71e5cec63e32cfa8a45R26